traits/etsconfig/tests/test_etsconfig.py

```diff
@@ -146,6 +147,21 @@ def test_set_application_data(self):
         self.ETSConfig.application_data = old
         self.assertEqual(old, self.ETSConfig.application_data)
 
+    def test_delete_application_data(self):
+        del self.ETSConfig.application_data
```

Comment:
Can we strengthen this test (and similar tests) at all? If we're expecting some observable side-effect from deleting `application_data`, we ought to be able to write a test that verifies that side-effect.